overlord/servicestate/quota_handlers.go

```diff
@@ -546,25 +638,14 @@ func ensureSnapServicesForGroup(st *state.State, t *state.Task, grp *quota.Group
 	return appsToRestartBySnap, nil
 }
 
-// restartSnapServices is used to restart the services for each snap
-// that was newly moved into a quota group iterate in a sorted order
-// over the snaps to restart their apps for easy tests.
-func restartSnapServices(st *state.State, t *state.Task, appsToRestartBySnap map[*snap.Info][]*snap.AppInfo, perfTimings *timings.Timings) error {
+// restartSnapServices is used to restart the services for snaps that
+// have been modified. Snaps and services are sorted before they are
+// restarted to provide a consistent ordering of restarts to be testable.
+func restartSnapServices(st *state.State, appsToRestartBySnap map[*snap.Info][]*snap.AppInfo) error {
```

Comment:
maybe for a follow up but as I said maybe this could be merged with the only caller now?

Reply:
my personal opinion is that it's better as-is, I like that functions are single responsibility for better maintenance, and also makes the intentions of the functions clearer.

Reply:
my worry is that because it doesn't do anything with the profiles is not as useful as a standalone function anymore, and we explain this problem only in its caller and not here